crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -311,25 +330,42 @@ impl WaylandWindowStatePtr {
     pub fn handle_xdg_surface_event(&self, event: xdg_surface::Event) {
         match event {
             xdg_surface::Event::Configure { serial } => {
+                {
+                    let mut state = self.state.borrow_mut();
+                    if let Some(window_controls) = state.in_progress_window_controls.take() {
+                        state.window_controls = window_controls;
+
+                        drop(state);
+                        let mut callbacks = self.callbacks.borrow_mut();
+                        if let Some(appearance_changed) = callbacks.appearance_changed.as_mut() {
+                            appearance_changed();
+                        }
+                    }
+                }
                 {
                     let mut state = self.state.borrow_mut();
 
                     if let Some(mut configure) = state.in_progress_configure.take() {
                         let got_unmaximized = state.maximized && !configure.maximized;
                         state.fullscreen = configure.fullscreen;
                         state.maximized = configure.maximized;
+                        state.tiling = configure.tiling;
+                        configure.size = compute_outer_size(
```

Comment:
I believe the `compute_outer_size` call isn't needed when maximizing?

Reply:
Changing the code to the following fixes maximization/unmaximization on Wayland:
```suggestion
                        configure.size = if configure.maximized {
                            configure.size
                        } else {
                            compute_outer_size(
                                state.window_bounds,
                                state.client_area,
                                configure.size,
                            )
                        };
```